core/Period/Factory.php

```diff
@@ -168,8 +168,11 @@ public static function makePeriodFromQueryParams($timezone, $period, $date)
         } else {
             if (!($date instanceof Date)) {
                 if (preg_match('/^(now|today|yesterday|yesterdaySameTime|last[ -]?(?:week|month|year))$/i', $date)) {
-                    // calculate the date taking the current time into account
-                    $date = Date::factory(Date::factory($date)->setTime(date('H:i:s', Date::getNowTimestamp()))->getTimestamp(), $timezone)->getDatetime();
+                    // most magic keywords will automatically set the time to 00:00:00
+                    // we need to set the time manually, so the timezone can be applied correctly
+                    $timestampWithCurrentTime = Date::factory($date)->setTime(date('H:i:s', Date::getNowTimestamp()))->getTimestamp();
```

Comment:
Now that I think about it, I think this is what the `Date::factoryInTimezone` method is meant to achieve. Can you check if this can be used instead? It's meant to retain the time of day before converting to the timezone. If it works, then the code will be simpler and we won't need to do something like this everywhere. Otherwise, feel free to merge.

Reply:
Good hint. Seems I missed that method for some reason. Using it does the trick as well. At least the tests are still passing.